packages/frontend/src/components/common/ActionCardList.tsx

```diff
@@ -76,13 +76,14 @@ const ActionCardList = <T extends { uuid: string; name: string }>({
                     ModalContent={ModalContent}
                 />
             )}
-            {actionState.actionType === ActionTypeModal.DELETE && (
-                <DeleteActionModal
-                    useActionModalState={[actionState, setActionState]}
-                    useDelete={useDelete}
-                    ModalContent={ModalContent}
-                />
-            )}
+            {actionState.actionType === ActionTypeModal.DELETE &&
+                actionState.data && (
+                    <DeleteDashboardModal
+                        refresh={new Date().getTime()}
```

Comment:
https://github.com/lightdash/lightdash/pull/1719#issuecomment-1092796843 explained on my previous comment

Reply:
@rephus I don't think I completely understand, can you make a short vid of the interaction someone would take in the UI to cause "model being retriggered again from the `ActionCardList`" ?

Reply:
sure, so, the first time I click on `delete` the modal is loaded the first time, and if I click cancel then `isOpen` state is set to False. However, without the `refresh` hack, if I click on the `same` dashboard a second time, the modal will not open because isOpen state is still false. There is another edge case that I fixed by using this, that happens when the dashboard is deleted (prevents the modal from showing twice) 

![Peek 2022-04-12 08-22](https://user-images.githubusercontent.com/1983672/162894319-a2994802-b1c7-404e-bbc2-9993eaed5d35.gif)

Reply:
@rephus could this be solved by pulling the state outside of the component? So the open state is controlled:

```typescript
                    <DeleteDashboardModal
                        isOpen={
                            actionState.actionType === ActionTypeModal.DELETE
                        }
                        onClose={() =>
                            setActionState({
                                actionType: ActionTypeModal.CLOSE,
                            })
                        }
                        uuid={actionState.data.uuid}
                        name={actionState.data.name}
```